library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c

```diff
@@ -193,6 +193,12 @@ static libspdm_return_t libspdm_try_negotiate_algorithms(libspdm_context_t *spdm
         status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
         goto receive_done;
     }
+    if (spdm_response->header.spdm_version >= SPDM_MESSAGE_VERSION_12) {
+        if (!libspdm_onehot0(spdm_response->other_params_support)) {
```

Comment:
it should be `if (!libspdm_onehot0(spdm_response->other_params_support & 0xF)) {`

The upper bit are reserved and should be ignored.

Reply:
The line 486~488 check the opaque data format must be **OpaqueDataFmt1**, but this PR is to check **no more than one bit** shall be set for the opaque data format.
https://github.com/DMTF/libspdm/blob/befd3a8c41427feb28719db4324ec2f82ed62790/library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c#L485-L488

Reply:
If it is only Fmt1, that already meets no more than one bit requirement.

I dont think more check is needed.

Reply:
ok, no more check is needed, and just add unit test for OtherParamsSelection has more than one bit set.